src/main/java/com/rackspacecloud/metrics/ingestionservice/influxdb/InfluxDBHelper.java

```diff
@@ -177,34 +177,25 @@ public InfluxDBFactory getInfluxDBFactory() {
      * @param tenantId
      * @param measurement
      * @return
-     * @throws Exception
      */
-    private TenantRoutes getTenantRoutes(String tenantId, String measurement) throws Exception {
+    private TenantRoutes getTenantRoutes(String tenantId, String measurement) {
         TenantRoutes tenantRoutes;
 
-        try {
-            tenantRoutes = routeProvider.getRoute(tenantId, measurement, restTemplate);
-        }
-        catch(Exception e) {
-            String errMsg = String.format(
-                    "Failed to get routes for tenantId [%s] and measurement [%s]", tenantId, measurement);
-            log.error(errMsg, e);
-            throw new Exception(errMsg, e);
-        }
+        tenantRoutes = routeProvider.getRoute(tenantId, measurement, restTemplate);
 
-        if(tenantRoutes == null) throw new Exception("tenantRoutes is null.");
+        if(tenantRoutes == null) throw new RouteNotFoundException(tenantId);
         return tenantRoutes;
     }
 
-    private boolean databaseExists(final String databaseName, final String baseUrl) throws Exception {
+    private boolean databaseExists(final String databaseName, final String baseUrl) {
         String queryString = "SHOW DATABASES";
 
         InfluxDB influxDB = getInfluxDBClient(baseUrl);
 
         if(influxDB == null) {
             String errorMessage = String.format("Couldn't get InfluxDB instance for URL [%s].", baseUrl);
             log.error(errorMessage);
-            throw new Exception(errorMessage);
+            throw new NullPointerException(errorMessage);
```

Comment:
I made this a null pointer exception because there was no instance for the URL. I am not clear if that means there is no route or if it means something else which is why I didn't make this a NoRouteException, but I am not happy with NullPointerException either. Was kind of hoping I could get some suggestions from you guys on this one.

Reply:
`IllegalStateException` is a good one for these general purpose cases where some internal expectation went wrong.

...however, looking into that call path it appears to be impossible for a null value to even get returned since the final call is

https://github.com/influxdata/influxdb-java/blob/influxdb-java-2.14/src/main/java/org/influxdb/InfluxDBFactory.java#L28-L31

Reply:
So you are good with me just removing this chunk of code then?

Reply:
Yes